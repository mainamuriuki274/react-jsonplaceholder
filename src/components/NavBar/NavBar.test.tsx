import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { NavBar } from './index';

describe('NavBar Component', () => {
  it('it renders', () => {
    render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>,
    );
    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('Users')).toBeInTheDocument();
    expect(screen.getByText('Posts')).toBeInTheDocument();
  });
  it('it matches snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <NavBar />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
