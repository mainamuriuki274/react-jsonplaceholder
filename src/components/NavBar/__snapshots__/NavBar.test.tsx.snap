// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component it matches snapshot 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark sticky-top"
>
  <div
    className="container"
  >
    <span
      className="navbar-brand"
    >
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Savannah Informatics
      </a>
    </span>
    <div
      className="justify-content-end navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="link"
        href="/"
        id="RouterNavLink"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="link"
        href="/users"
        id="RouterNavLink"
        onClick={[Function]}
      >
        Users
      </a>
      <a
        className="link"
        href="/posts"
        onClick={[Function]}
      >
        Posts
      </a>
    </div>
  </div>
</nav>
`;

exports[`NavBar Component it matches snapshot 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark sticky-top"
>
  <div
    className="container"
  >
    <span
      className="navbar-brand"
    >
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Savannah Informatics
      </a>
    </span>
    <div
      className="justify-content-end navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="link"
        href="/"
        id="RouterNavLink"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="link"
        href="/users"
        id="RouterNavLink"
        onClick={[Function]}
      >
        Users
      </a>
      <a
        className="link"
        href="/posts"
        onClick={[Function]}
      >
        Posts
      </a>
    </div>
  </div>
</nav>
`;
